android/BOINC/app/src/main/java/edu/berkeley/boinc/client/NoticeNotification.kt

```diff
@@ -0,0 +1,215 @@
+/*
+  This file is part of BOINC.
+  https://boinc.berkeley.edu
+  Copyright (C) 2022 University of California
+  
+  BOINC is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License
+  as published by the Free Software Foundation,
+  either version 3 of the License, or (at your option) any later version.
+  
+  BOINC is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+  
+  You should have received a copy of the GNU Lesser General Public License
+  along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+**/
+package edu.berkeley.boinc.client
+
+import android.app.Notification
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.net.Uri
+import android.os.Build.VERSION
+import android.os.Build.VERSION_CODES
+import androidx.annotation.RequiresApi
+import androidx.annotation.VisibleForTesting
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationCompat.BigTextStyle
+import androidx.core.app.NotificationCompat.Builder
+import androidx.core.app.NotificationCompat.InboxStyle
+import androidx.core.app.NotificationManagerCompat
+import edu.berkeley.boinc.BOINCActivity
+import edu.berkeley.boinc.R.drawable
+import edu.berkeley.boinc.R.integer
+import edu.berkeley.boinc.R.mipmap
+import edu.berkeley.boinc.R.plurals
+import edu.berkeley.boinc.R.string
+import edu.berkeley.boinc.rpc.Notice
+import edu.berkeley.boinc.utils.Logging.Category.CLIENT
+import edu.berkeley.boinc.utils.Logging.logException
+import edu.berkeley.boinc.utils.getBitmapFromVectorDrawable
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@RequiresApi(VERSION_CODES.M)
```

Comment:
We support Android versions starting from Android 4.1 (API level 16).
If this set, this will not work for any that is older than Android 6.0 (API level 23).
So you should either find a way to fix that warning without increasing making ANdroid 6.0 as a minimal supported version or just rollback and leave it as-is (the first in preferred).

Reply:
I can add the below code, but a deprecated warning won't go away
`        
contentIntent = if (VERSION.SDK_INT >= VERSION_CODES.M)	{	
PendingIntent.getActivity(context, 0, intent,PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_UPDATE_CURRENT) } else { PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT) }
`

Reply:
Then just rollback your last commit and it should be ok

Reply:
And add this annotation at the top of this class:
```
@SuppressLint("UnspecifiedImmutableFlag")
```